src/program/downloaders/alldebrid.py

```diff
@@ -51,594 +34,185 @@ def validate(self) -> bool:
-                                for subfile in file.e:
-                                    if isinstance(item, Movie) and self._is_wanted_movie(subfile, item) or isinstance(item, Episode) and self._is_wanted_episode(subfile, item):
-                                        item.set("file", subfile.n)
-                                        break
-            if not item.folder or not item.alternative_folder or not item.file:
-                logger.error(f"Missing folder or alternative_folder or file for item: {item.log_string}")
-                return
-    
-        if isinstance(item, Season) and item.folder:
-            for episode in item.episodes:
-                if episode.file and not episode.folder:
-                    episode.set("folder", item.folder)
-    
-        if isinstance(item, Show) and item.folder:
-            for season in item.seasons:
-                for episode in season.episodes:
-                    if episode.file and not episode.folder:
-                        episode.set("folder", item.folder)
-    
-        # Handle nested files in the links
-        for link in magnet_info.links:
-            if hasattr(link, "files"):
-                for file in link.files:
-                    if isinstance(file, SimpleNamespace) and hasattr(file, "e"):
-                        for subfile in file.e:
-                            if isinstance(item, Season) and self._is_wanted_season(link.files, item) or isinstance(item, Show) and self._is_wanted_show(link.files, item):
-                                break
-    
-        if isinstance(item, Season) and item.folder:
-            for episode in item.episodes:
-                if episode.file and not episode.folder:
-                    episode.set("folder", item.folder)
-    
-        if isinstance(item, Show) and item.folder:
-            for season in item.seasons:
-                for episode in season.episodes:
-                    if episode.file and not episode.folder:
-                        episode.set("folder", item.folder)
-    
-        # Handle nested files in the links
-        for link in magnet_info.links:
-            if hasattr(link, "files"):
-                for file in link.files:
-                    if isinstance(file, SimpleNamespace) and hasattr(file, "e"):
-                        for subfile in file.e:
-                            if isinstance(item, Season) and self._is_wanted_season(link.files, item) or isinstance(item, Show) and self._is_wanted_show(link.files, item):
-                                break
-
-    ### API Methods for All-Debrid below
-    def add_magnet(self, item: MediaItem) -> str:
-        """Add magnet link to All-Debrid"""
-        if not item.active_stream.get("hash"):
-            logger.error(f"No active stream or hash found for {item.log_string}")
-            return None
-    
+def walk_alldebrid_files(files: dict) -> (str, int):
```

Comment:
Seems like `get_cached_container` just wants a list of files, so that's what I've given it - but again it's hard to tell without seeing some RealDebrid API responses.

Reply:
here is one example of  a cached container on RD via the `instantAvailability` endpoint.
- `/torrents/instantAvailability/47E4B344A13485BED48798A2A42C74FDFF14F3F0`

```json
{
    "47e4b344a13485bed48798a2a42c74fdff14f3f0": {
        "rd": [
            {
                "4": {
                    "filename": "Wolfs.2024.2160p.ATVP.WEB-DL.DV.HDR10.PLUS.ENG.LATINO.DDP5.1.Atmos.H265.MP4-BEN.THE.MEN.mp4",
                    "filesize": 21827787647
                }
            },
            {
                "1": {
                    "filename": "Downloaded From The Pirate Bay.txt",
                    "filesize": 409
                },
                "2": {
                    "filename": "Downloaded From glodls.to.txt",
                    "filesize": 384
                },
                "3": {
                    "filename": "Downloaded From torrentgalaxy.to.txt",
                    "filesize": 981
                },
                "4": {
                    "filename": "Wolfs.2024.2160p.ATVP.WEB-DL.DV.HDR10.PLUS.ENG.LATINO.DDP5.1.Atmos.H265.MP4-BEN.THE.MEN.mp4",
                    "filesize": 21827787647
                }
            }
        ]
    }
}
```

Reply:
Thanks! Each block seems to be the same torrent, just with different subsets of files exposed?

I've been running this downloader for about a day and it really seems to struggle to create symlinks for a lot of torrents, but it does work sometimes even when the content is nested. Looking at the symlink library I think what I'm doing here is correct. Directories are discarded since the symlinker only seems to care about file name